test/qml/tst_editorwidgets.qml

```diff
@@ -2,45 +2,356 @@ import QtQuick 2.3
 import QtTest 1.0
 
 import org.qfield 1.0
-
 import Theme 1.0
 
 import "../../src/qml/editorwidgets" as EditorWidgets
 
 TestCase {
-    name: "EditorWidgets"
+  name: "EditorWidgets"
+
+  EditorWidgets.TextEdit {
+    id: textEdit
+    property string value: "DEFAULT_VALUE"
+    property var config: undefined
+    property bool isEnabled: true
+  }
+
+  EditorWidgets.Range {
+    id: range
+    property real value: default_value
+    property var config: undefined
+    property bool isEnabled: true
+
+    readonly property real default_value: 999
+  }
+
+  EditorWidgets.DateTime {
+    id: dateTime
+    property string value: "2022-01-01"
+    property var config: undefined
+    property var field: undefined
+    property bool isEnabled: true
+  }
+
+  EditorWidgets.CheckBox {
+    id: checkBox
+    property bool value: true
+    property var config: undefined
+    property var field: undefined
+  }
+
+  EditorWidgets.ValueMap {
+    id: valueMap
+    property var value: undefined
+    property var config: undefined
+    property var field: undefined
+
+    property var currentLayer: undefined
+
+    Item {
+      id: currentLayerTrue
+      function customProperty(value) {
+        return true
+      }
+    }
+
+    Item {
+      id: currentLayerFalse
+      function customProperty(value) {
+        return false
+      }
+    }
+  }
+
+
+  /**
+   * Test case for textEdit widget
+   *
+   * This function tests the textEdit widget's functionality by setting its config to allow multiline and use html,
+   * setting its value to "six", verifying that it contains "six", resetting its config to disable multiline and html,
+   * setting its value to "seven", and comparing its children's texts with "six" and "seven".
+   */
+  function test_01_textEdit() {
+    const textReadonlyValue = textEdit.children[0]
+    const textField = textEdit.children[1]
+    const textArea = textEdit.children[2]
```

Comment:
We should change this to something a bit more flexible, such as relying on objectName (and creating a small utility function to find / return an item by objectName). Not for this PR though.

Reply:
Yes, i agree.